python/ray/serve/_private/http_proxy.py

```diff
@@ -998,7 +992,11 @@ async def check_health(self):
         """No-op method to check on the health of the HTTP Proxy.
         Make sure the async event loop is not blocked.
         """
-        pass
+        if self.app.is_drained():
+            logger.info(
+                f"The proxy actor on node {self.app._node_id} is drained. Exiting..."
+            )
+            ray.actor.exit_actor()
```

Comment:
this still looks pretty hacky to me... IMO the more robust solution would be to return a status code from this health check, then have the controller make the decision to remove the proxy

I won't block the PR on this change if it's much harder to model it that way, but having an implicit signal of the actor killing itself doesn't sit very well with me